harmony/http.py

```diff
@@ -67,6 +73,56 @@ def localhost_url(url, local_hostname):
     return url.replace('localhost', local_hostname)
 
 
+def _mount_retry(session, total_retries=DEFAULT_TOTAL_RETRIES, backoff_factor=0.2):
+    """Mount a retry adapter to a requests session.
+
+    Parameters
+    ----------
+    session : requests.Session
+        The session that will have a retry adapter mounted to it.
+    total_retries: int
+        Upper limit on the number of times to retry the request
+    backoff_factor: float
+        Factor used to determine backoff/sleep time between executions:
+        backoff = {backoff factor} * (2 ** ({number of total retries} - 1))
```

Comment:
I'm finding the math really tough to follow - does this mean that with 10 retries, and a backup factor of 0.2 we will retry at:
0.2, 0.4, 0.8, 1.6, 3.2, 6.4, 12.8, 25.6, 51.2 seconds?

If so that sounds too fast early on and not quite long enough to recover from something like a 5 minute outage. Once we have this sort of figured out can you put a table like above in the comments so it's clear what the default retry behavior is?

Reply:
I realized I should put some recommended values - maybe:
5, 10, 20, 40, 80, 160, 320?

That would give a total of a little over 10 minutes, which would be nice for errors that could succeed on retry, but annoying for something that will always fail.

Reply:
I _might_ also be able to set Retry.DEFAULT_BACKOFF_MAX.

Reply:
OK so after some more logging it's doing multiple sleeps between each retry.

backoff_factor=5:

sleep(0);, 
sleep(0); sleep(10);, 
sleep(0); sleep(10); sleep(20);
sleep(0); sleep(10); sleep(20); sleep(40);

Reply:
If we do a backoff factor of 2, we'll get 7 retries and a total of 8 minutes of sleep (0, 4, 12, 28, 60, 124, 252). Would that be OK?